wallet/wallet-controller/src/synced_controller.rs

```diff
@@ -630,6 +649,154 @@ impl<'a, T: NodeInterface, W: WalletEvents> SyncedController<'a, T, W> {
         Ok((req, fees))
     }
 
+    /// Create an unsigned transaction for transfer of tokens to the specified destinations.
+    ///
+    /// The inputs for the transfer are randomly selected from the provided `inputs` until
+    /// target amounts are satisfied.
+    ///
+    /// The function will automatically select coin utxos to pay fees from (note that `inputs`
+    /// is not supposed to contain coin utxos; if it does, there is no guarantee that exactly
+    /// those utxos will be selected to pay the fee).
+    ///
+    /// The change will be sent to addresses specified in `change_addresses`.
+    /// If there is no entry in `change_addresses` for a certain token, the change will be sent
+    /// to the first unused address in the wallet.
+    /// If there is no entry in `change_addresses` for coins, the destination for the change
+    /// from the fee payment will be taken from one of the existing coin utxos.
+    // TODO: this discrepancy between tokens/coins fee handling is a bit ugly, it's better to unify it.
+    // Note: the reason for this specific fee change behavior is that this function is called from
+    // `make_tx_to_send_tokens_from_multisig_address`, which is supposed to be used in automated
+    // scenarios; issuing a new address for each call seems redundant in such a case.
+    // The token change being sent to first unused address doesn't have any particular reason;
+    // it's just how this function's callee behaves by default.
+    pub async fn make_unsigned_tx_to_send_tokens_to_addresses(
+        &mut self,
+        inputs: Vec<(UtxoOutPoint, TxOutput)>,
+        outputs: BTreeMap<TokenId, Vec<GenericTxOutput>>,
```

Comment:
I think `GenericTxOutput` adds unnecessary complexity. Providing `BtreeMap<TokenId, (DecimalAmount, Destination)>` would do the job without introducing another TxOutput analog. In terms of wallet `send_*` is always `TxOutput::Transfer` and I don't think we would need anything else.

Reply:
> So I'll make it a struct and rename to something like CurrencyTransfer. Also, GenericTxTokenOutput will become TokenTransfer

I've named them `GenericCurrencyTransfer` and `GenericTokenTransfer` respectively.